hawc/apps/assessment/views.py

```diff
@@ -855,3 +857,92 @@ def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
         context = super().get_context_data(**kwargs)
         context["content_types"] = self.get_cts()
         return context
+
+
+class PublishedItemsChecklist(HtmxViewSet):
+    actions = {"list", "update_item"}
+    parent_actions = {"list", "update_item"}
+    parent_model = models.Assessment
+
+    @action(permission=can_edit, htmx_only=False)
+    def list(self, request: HttpRequest, *args, **kwargs):
+        return render(
+            request,
+            "assessment/published_items.html",
+            self.get_list_context_data(self.request.user, self.request.item.assessment),
+        )
+
+    @action(permission=can_edit, methods={"post"})
+    def update_item(self, request: HttpRequest, *args, **kwargs):
+        instance = self.get_instance(request.item, kwargs["type"], kwargs["object_id"])
+        self.perform_update(request, instance)
+        return render(
+            request,
+            "assessment/fragments/publish_item_td.html",
+            {"name": kwargs["type"], "object": instance, "assessment": request.item.assessment},
+        )
+
+    def get_instance(self, item, type: str, object_id: int):
+        lookup = {
+            "study": apps.get_model("study", "Study"),
+            "visual": apps.get_model("summary", "Visual"),
+            "datapivot": apps.get_model("summary", "DataPivot"),
+            "dataset": apps.get_model("assessment", "Dataset"),
+            "summarytable": apps.get_model("summary", "SummaryTable"),
+            "attachment": apps.get_model("assessment", "Attachment"),
+        }
+        Model = lookup.get(type)
+        if not Model:
+            raise Http404()
+        return get_object_or_404(Model.objects.filter(id=object_id))
```

Comment:
changed this to accommodate for attachments--does it matter if we do not check for the correct assessment relation?

Reply:
can you elaborate? I'm not following with the relation comment

Reply:
yeah, this line used to be `return get_object_or_404(Model.objects.filter(assessment=item.assessment), id=object_id)` but I took out the assessment part since attachments are not directly related to assessments

Reply:
I feel like it should function the same with only the object id but I wanted to double check

Reply:
without that check, couldn't someone change content across assessments? That is, they could pass an assessment id where they have write access, and an object id that they don't have access to

Reply:
@casey1173 I dont think this was changed right? Dont we need some sort of permissions check to ensure that the thing you are requesting to change is part of the assessment?